src/core/components/files.js

```diff
@@ -86,23 +177,40 @@ module.exports = function files (self) {
       )
     }),
 
-    cat: promisify((ipfsPath, callback) => {
-      if (typeof ipfsPath === 'function') {
-        return callback(new Error('You must supply an ipfsPath'))
-      }
+    addReadableStream: (options) => {
+      options = options || {}
+
+      const p = pushable()
+      const s = pull(
+        p,
+        _addPullStream(options)
+      )
 
+      const retStream = new AddHelper(s, p)
+
+      retStream.once('finish', () => p.end())
+
+      return retStream
+    },
+
+    addPullStream: _addPullStream,
+
+    cat: promisify((ipfsPath, callback) => {
+      const p = _catPullStream(ipfsPath)
       pull(
-        exporter(ipfsPath, self._ipldResolver),
-        pull.collect((err, files) => {
-          if (err) {
-            return callback(err)
-          }
-          if (!files || !files.length) return callback(new Error('No such file'))
-          callback(null, toStream.source(files[files.length - 1].content))
-        })
+        p,
+        pull.concat(callback)
       )
     }),
```

Comment:
@pgte the .cat for a large file (15Mb) is failing with:

<img width="603" alt="image" src="https://user-images.githubusercontent.com/1211152/32946666-5c8e6d94-cb90-11e7-8308-c1833258e6c8.png">

Seems that some of the streams is giving up too early. Could you help me figuring this one out?

Reply:
Think I got it, I can't use concat to concat buffers https://github.com/pull-stream/pull-stream/blob/master/sinks/concat.js